modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer.js

```diff
@@ -116,53 +95,16 @@ export default class MultiIconLayer extends IconLayer {
     });
   }
 
-  calculateInstanceOffsets(attribute, {startRow, endRow}) {
-    const {
-      data,
-      iconMapping,
-      getIcon,
-      getAnchorX,
-      getAnchorY,
-      getParagraphSize,
-      getRowSize,
-      getOffsets
-    } = this.props;
-    const {value, size} = attribute;
-    let i = startRow * size;
-    const {iterable} = createIterable(data, startRow, endRow);
-
-    for (const object of iterable) {
-      const icon = getIcon(object);
-      const rect = iconMapping[icon] || {};
-      const [width, height] = getParagraphSize(object);
-      const [rowWidth] = getRowSize(object);
-      const [offsetX, offsetY] = getOffsets(object);
-      const anchorX = getAnchorX(object);
-      const anchorY = getAnchorY(object);
-
-      // For a multi-line object, offset in x-direction needs consider
-      // the row offset in the paragraph and the object offset in the row
-      const rowOffset = ((1 - anchorX) * (width - rowWidth)) / 2;
-      value[i++] = ((anchorX - 1) * width) / 2 + rowOffset + rect.width / 2 + offsetX || 0;
-      value[i++] = ((anchorY - 1) * height) / 2 + rect.height / 2 + offsetY || 0;
-    }
+  getInstanceOffset(icons) {
+    return Array.from(icons).map(icon => super.getInstanceOffset(icon));
   }
 
-  calculateInstancePickingColors(attribute, {startRow, endRow}) {
-    const {data, getPickingIndex} = this.props;
-    const {value, size} = attribute;
-    let i = startRow * size;
-    const pickingColor = [];
-    const {iterable} = createIterable(data, startRow, endRow);
-
-    for (const point of iterable) {
-      const index = getPickingIndex(point);
-      this.encodePickingColor(index, pickingColor);
+  getInstanceColorMode(icons) {
+    return 1; // mask
```

Comment:
define a constant.

Reply:
It is not needed elsewhere.

Reply:
Context: colorMode is used as a boolean flag (`true` for masked, `false` for not) but because it is an attribute we have to fill the array with an int (`1` and `0`).